src/test/java/com/unzer/payment/service/UrlUtilTest.java

```diff
@@ -0,0 +1,74 @@
+package com.unzer.payment.service;
+
+/*-
+ * #%L
+ * Unzer Java SDK
+ * %%
+ * Copyright (C) 2020 Unzer E-Com GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+
+import com.unzer.payment.Unzer;
+import com.unzer.payment.business.AbstractPaymentTest;
+import com.unzer.payment.business.paymenttypes.InstallmentSecuredRatePlan;
+import com.unzer.payment.communication.HttpCommunicationException;
+import com.unzer.payment.paymenttypes.InvoiceSecured;
+import com.unzer.payment.paymenttypes.SepaDirectDebitSecured;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class UrlUtilTest extends AbstractPaymentTest {
+
+    @Test
+    public void testGetUrlForPaymentTypeInvoiceSecured() throws HttpCommunicationException {
+        String minimalRestUrl = "https://api.unzer.com/v1/types";
+        String maximumRestUrl = "https://api.unzer.com/v1/types/invoice-secured";
+
+        Unzer unzer = getUnzer();
+        UrlUtil urlUtil = new UrlUtil(unzer.getEndPoint());
+        InvoiceSecured invoiceSecured = createPaymentTypeInvoiceSecured();
+
+        assertEquals(String.format("%s/%s", minimalRestUrl, invoiceSecured.getId()), urlUtil.getHttpGetUrl(invoiceSecured.getId()));
+        assertEquals(String.format("%s/%s", maximumRestUrl, invoiceSecured.getId()), urlUtil.getHttpGetUrl(invoiceSecured, invoiceSecured.getId()));
+    }
+
+    @Test
+    public void testGetUrlForPaymentTypeInstallmentSecured() throws HttpCommunicationException {
+        String minimalRestUrl = "https://api.unzer.com/v1/types";
+        String maximumRestUrl = "https://api.unzer.com/v1/types/invoice-secured";
+
+        Unzer unzer = getUnzer();
+        UrlUtil urlUtil = new UrlUtil(unzer.getEndPoint());
+        InstallmentSecuredRatePlan installmentSecuredRatePlan = createPaymentTypeInstallmentSecured();
+
+        assertEquals(String.format("%s/%s", minimalRestUrl, installmentSecuredRatePlan.getId()), urlUtil.getHttpGetUrl(installmentSecuredRatePlan.getId()));
+        assertEquals(String.format("%s/%s", maximumRestUrl, installmentSecuredRatePlan.getId()), urlUtil.getHttpGetUrl(installmentSecuredRatePlan, installmentSecuredRatePlan.getId()));
```

Comment:
I can't see how this should work, since the defined URL shouldn't fit to the type.
Also the test is called ...`InstallmentSecured` therefore i would also expect that type here and not an `InstallmentSecuredRatePlan`.

Reply:
Odly the test doesn't ran on my side because of the missing configuration in the Java-SDK Keypair. I will try it with the PHP-Keys to check it.

Reply:
Have changed it and ran the tests with the PHP-SDK-Keys and it worked!